### ISSUE TITLE
CI does not update private Beta registry if `Skip.PublishVcpkg = true`.

### ISSUE THREAD

**Issue by antkmsft** [Azure/azure-sdk-for-cpp#3700]
This is not exactly what we want.
When `Skip.PublishVcpkg = true`, we don't want the PR for the vcpkg repo. But we do want to update our private Beta registry.

Expected: Beta registry gets updated regardless of whether `Skip.PublishVcpkg` is true or not.

**Comment by danieljurek** [comment]
Thanks for filing @antkmsft, this is a bug. 

The line here: https://github.com/Azure/azure-sdk-for-cpp/blame/main/eng/pipelines/templates/steps/vcpkg-publish.yml#L49

Should not be indented as deeply. 

The pipeline run is correctly identifying that a package should not be published to vcpkg in this case: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1483258&view=logs&j=3dbdc0e8-0699-52f6-1880-be74b44e10fc&t=589fe240-e4d3-576e-326a-612f68def396
![image](https://user-images.githubusercontent.com/2158838/171907106-d0946f72-c07d-4486-bf2b-3be9175af6d3.png)

And yet the step was running anyway. 

Once this is indented properly I expect skipping behavior to work correctly.

**Comment by antkmsft** [comment]
Thank you @danieljurek! So, what will be the correct guidance when releasing post-GA Betas, once this is fixed - with `Skip.PublishVcpkg` being true or false?

cc @Jinming-Hu

**Comment by danieljurek** [comment]
Only set `Skip.PublishVcpkg` to `true` if don't want to publish to vcpkg AND vcpkg beta. All vcpkg publishing is skipped. 

If you leave `Skip.PublishVcpkg` un-set then the vcpkg publishing process will look like: 

| Version publishing | Version already published to main vcpkg | Publish to main vcpkg | Publish to vcpkg betas | 
| -------------------- | --------------------------------------------- | ----------------------- | ------------------------- | 
| 1.0.0-beta.1           | \<none\>                                                    | Yes                             | Yes                                | 
| 1.0.0-beta.2           | 1.0.0-beta.1                                                | Yes                              | Yes                                | 
| 1.0.0                      | 1.0.0-beta.2                                                | Yes                               | Yes                               | 
| 1.1.0-beta.1          | 1.0.0                                                            | NO                               | Yes                               | 
| 1.1.0                      | 1.0.0                                                           | Yes                               | Yes                               | 

In short, we always publish to vcpkg betas and we publish main vcpkg UNLESS the a version already on vcpkg main is a GA version and the version being released is a beta. That is, beta versions of packages do not release over GA packages on the main vcpkg... Customers should not be forced to use a beta package when a GA package is available.

**Comment by Jinming-Hu** [comment]
@danieljurek If I want to ship GA version for storage-common package and beta version for storage-blobs package, do I have to do that in two separate pipelines, or can I do it in one pipeline?

**Comment by danieljurek** [comment]
@Jinming-Hu -- You can do this in the same pipeline execution. Packages are evaluated independently to determine whether to publish to main vcpkg.
